sdk/inc/azure/core/az_http.h

```diff
@@ -31,8 +31,6 @@
  */
 typedef enum
 {
-  AZ_HTTP_STATUS_CODE_NONE = 0,
```

Comment:
I thought we decided to keep this because it was needed for something.

Reply:
It is not used anywhere. I thought it was used for the scenario when the cable gets unplugged, but it was the old design, now az_http_send_reqyes would return non-AZ_OK result.
But... You know what, let's keep it. If you have response default-initialized, this is what you re going to have there. Not that we have to have a constant defined for the 0 value (enums still can take "undefined" random integer values), but I think it is better to have it.

Reply:
Do you think it is worth documenting it to make clear that is what it's purpose is?

I think overloading the 0 HTTP_STATUS_CODE might confuse folks into thinking there is an actual status code of value zero. We may want to avoid that and remove it.

Reply:
To be fair, AZ_HTTP_STATUS_CODE_END_OF_LIST doesn't map to a HTTP status code either.

Reply:
Jeffrey told me in Teams: `Sure, let's keep both. Before a response comes back, the status code will be NONE. And, if some network error occurs, then the status code will also be NONE.`

Reply:
Wouldn't a network error set the code to some 5xx error like 503, 523, or 598? Why would it be `NONE` in that case?